### PR TITLE
[#2313] Porting IVC+ADD 4: Add first folding iteration example

### PR DESCRIPTION
**Pull request by volhovm**

### CODE REVIEW

**Review by fabrizio-m — APPROVED** [review]

**Comment by volhovm** [comment]
:tada:

File: ivc/tests/add.rs [hunk 1]
```diff
@@ -467,36 +470,152 @@ pub fn heavy_test_simple_add() {
...
+    let empty_lookups_app = BTreeMap::new();
+    let empty_lookups_ivc = BTreeMap::new();
+
+    // Witness one
+    for _i in 0..domain_size {
+        let a: Fp = Fp::rand(&mut rng);
+        let b: Fp = Fp::rand(&mut rng);
+        app_witness_one.write_column(AdditionColumn::A, &a);
+        app_witness_one.write_column(AdditionColumn::B, &b);
+        app_witness_one.write_column(AdditionColumn::C, &(a * a * b));
```
Comment (dannywillems) [inline ivc/tests/add.rs:495]: We're not proving an ADD anymore.

File: ivc/tests/add.rs [hunk 2]
```diff
@@ -467,36 +470,152 @@ pub fn heavy_test_simple_add() {
...
+    // want to intersperse them in a smarter way later. Our witness is
+    // Relation || dynamic.
+    let joint_witness_one: Vec<_> = proof_inputs_one
+        .evaluations
+        .clone()
+        .into_iter()
+        .chain(ivc_proof_inputs_0.evaluations.clone())
+        .collect();
+
+    for i in 0..10 {
```
Comment (dannywillems) [inline ivc/tests/add.rs:525]: Why?

Comment (volhovm) [reply ivc/tests/add.rs:525]: Just a sanity check that I used for debugging. Can be removed in the future.

**Pull request merged**